### ISSUE TITLE
where is the deployed cloudformation template?

### ISSUE THREAD

**Issue by cyrfer** [aws-amplify/amplify-cli#6045]
I want to know where is the Cloudformation template that is used by the Cloudformation service when deploying my app.

**Which Category is your question related to?**
Local Deployment

**Amplify CLI Version** 
4.37.0

**What AWS Services are you utilizing?**
Many.

**Provide additional details e.g. code snippets**
I customized one of the cloudformation templates. I added a stack parameter, I specified the parameter value in team-provider-info.json, and used the parameter as an ENV_VAR on my Lambda Function.

When I attempt to deploy, the local terminal output from `amplify push` and the cloudformation console show CF template syntax was the problem. The template in my `amplify/backend/function/foo` directory looks fine to me. I want to validate the template that is actually used for deployment. I examined the templates located at `s3://my-deployment-bucket/#current-cloud-backend.zip/function/foo` but it does not include the parameter code that I added.

Please point me to documentation that talks about where my template is sent when deploying.

**Comment by SwaySway** [comment]
The `backend` directory (https://docs.amplify.aws/cli/reference/files) is stored in your deployment bucket. The root stack would be `nested-cloudformation-stack.yml` and then the child stacks are located within `amplify-cfn-templates`.

**Comment by cyrfer** [comment]
`amplify-cfn-templates` THANK YOU! I overlooked this.

My template syntax issue was caused by the way I was trying to conditionally tack on a policy statement and an environment variable for a Lambda function.

The example below is the right way to do it. Instead of `{"Ref": "AWS::NoValue"}` (correct) I was using `"AWS::NoValue"` (incorrect).

```json
{
  "Fn::If": [
    "ShouldNotUseKmsKey",
    {
      "Ref": "AWS::NoValue"
    },
    {
      "Effect": "Allow",
      "Action": [
        "kms:*"
      ],
      "Resource": {
        "Ref": "KmsKeyId"
      }
    }
  ]
}
```

Side note, using `aws cloudformation validate-template --template-body file://foo.json` did not reveal the syntax errors reported by CF console.
